dts/bindings/usb/usb-ep.yaml

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (c) 2018, I-SENSE group of ICCS
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: USB Endpoints' properties
+id: USB-EP
+version: 0.1
+
+description: >
+    This binding gives number of endpoints that the USB hardware supports
+
+inherits:
+    !include usb.yaml
+
+properties:
+    num-bidir-endpoints:
+      type: int
+      category: required
+      description: Number of bi-directional endpoints supported by hardware
+                   (including EP0)
+      generation: define
+
+    num-in-endpoints:
+      type: int
+      category: optional
+      description: Number of IN endpoints supported by hardware
+                   (including EP0 IN)
+      generation: define
+
+    num-out-endpoints:
+      type: int
+      category: optional
+      description: Number of OUT endpoints supported by hardware
+                   (including EP0 OUT)
+      generation: define
+...
```

Comment:
I know this PR's been merged and it's a little late for this comment.
Been busy with other stuff and only now I realize there's a need for this sort of thing for PR #6256 to make it device tree compatible.

Here's a rough idea about nRF52840 USB:
* Any set of endpoint combination can be used, apart from control endpoints of course. (HW allows this sort of thing).
* The ISO endpoints are separate/dedicated and are optional from driver perspective (by SW design). Not any endpoint can be used as an ISO endpoint.
* Enabling ISO endpoint(s) requires additional buffer(s) to be allocated in the driver (by SW design).

 Now onto the questions:
1) Can we have num-isoin-endpoints and num-isoout-endpoints to describe their count separately?
2) Since not all the endpoints can be configured as ISO IN/OUT endpoint in Nordic's USB, having a property to describe the ISO endpoint(s)' number would be good which can be used by application (like hci_usb for example) in an agnostic manner. Although this is not required for certain chipsets which support ISO in any endpoint number, for chips like Nordic it would help really well. That way we can keep the application immune to any cluttering #ifdeffery to support different chips.
3) The way I understand num-bidir-endpoints is that it sort of describes the same thing as num-in-endpoints + num-out-endpoints. Although in certain cases, the in and out endpoint count need not be same, meaning some of them could be unidirectional. If so, can we just have the num-in-endpoint and num-out-endpoint properties alone? If num-bidir-endpoint is required then it may have to indicate which endpoint number is bidirectional.

Reply:
@sundar-subramaniyan Thanks for the comments.

> 1. Can we have num-isoin-endpoints and num-isoout-endpoints to describe their count separately?
>2. Since not all the endpoints can be configured as ISO IN/OUT endpoint in Nordic's USB, having a property to describe the ISO endpoint(s)' number would be good which can be used by application (like hci_usb for example) in an agnostic manner. Although this is not required for certain chipsets which support ISO in any endpoint number, for chips like Nordic it would help really well. That way we can keep the application immune to any cluttering #ifdeffery to support different chips.

You could add num-isoin-endpoints and num-isoout-endpoints in nordic,usb.yaml file. If other chipsets need them, we could move them later to usb-ep.yaml.

>3. The way I understand num-bidir-endpoints is that it sort of describes the same thing as num-in-endpoints + num-out-endpoints. Although in certain cases, the in and out endpoint count need not be same, meaning some of them could be unidirectional. If so, can we just have the num-in-endpoint and num-out-endpoint properties alone? If num-bidir-endpoint is required then it may have to indicate which endpoint number is bidirectional.

I added num-bidir-endpoints property as required because all USB devices have a bi-directional control endpoint (EP0). You could set num-bidir-endpoints property to 1 and use num-in-endpoints and num-out-endpoints for the rest.

Reply:
> You could add num-isoin-endpoints and num-isoout-endpoints in nordic,usb.yaml file. If other chipsets need them, we could move them later to usb-ep.yaml.
Sounds good.

> I added num-bidir-endpoints property as required because all USB devices have a bi-directional control endpoint (EP0). You could set num-bidir-endpoints property to 1 and use num-in-endpoints and num-out-endpoints for the rest.
Yes, for EP0 endpoint I can use bidir-endpoint. The asymmetric in/out configuration is used in some applications and the num-{in, out}-endpoint properties can be used respectively.

Thanks for the clarification.